phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java

```diff
@@ -233,16 +240,29 @@ public int run(String[] args) throws Exception {
       // Build and start the HttpServer
       server = builder.build();
       server.start();
+      registerToServiceProvider(hostname);
       runningLatch.countDown();
       server.join();
       return 0;
     } catch (Throwable t) {
       LOG.fatal("Unrecoverable service error. Shutting down.", t);
       this.t = t;
       return -1;
+    } finally {
+      deRegister();
     }
   }
 
+  private void registerToServiceProvider(String hostName) throws Exception {
+      PqsZookeeperConf pqsZookeeperConf = new PqsZookeeperConfImpl(getConf());
```

Comment:
Move any IT exercising loadbalancer functionality to the loadbalancer module :)

Tests in the queryserver module would have no load balancer implementation. PQS should pass gracefully in this case (e.g. not try to register with the loadbalancer because it would be null).